src/components/Editor/TableOfContents.vue

```diff
@@ -121,67 +169,69 @@ export default {
 			);
 		}
 
-		&--1 {
-			--padding-left: 0rem;
+		&.level1 {
+			--padding-inline-start: calc(0 * var(--level-padding));
 			font-weight: 600;
 			&:not(:nth-child(1)) {
 				margin-top: 0.5rem;
 			}
 		}
 
-		&--2 {
-			--padding-left: 1rem;
+		&.level2 {
+			--padding-inline-start: calc(1 * var(--level-padding));
 		}
 
-		&--3 {
-			--padding-left: 2rem;
+		&.level3 {
+			--padding-inline-start: calc(2 * var(--level-padding));
 		}
 
-		&--4 {
-			--padding-left: 3rem;
+		&.level4 {
+			--padding-inline-start: calc(3 * var(--level-padding));
 		}
 
-		&--5 {
-			--padding-left: 4rem;
+		&.level5 {
+			--padding-inline-start: calc(4 * var(--level-padding));
 		}
 
-		&--6 {
-			--padding-left: 5rem;
+		&.level6 {
+			--padding-inline-start: calc(5 * var(--level-padding));
```

Comment:
I think you could use an scss for loop here.
https://sass-lang.com/documentation/at-rules/control/for/